mteb/evaluation/evaluators/ClusteringEvaluator.py

```diff
@@ -26,7 +25,7 @@ def __call__(self, model):
 
         logger.info("Fitting Mini-Batch K-Means model...")
         clustering_model = sklearn.cluster.MiniBatchKMeans(
-            n_clusters=len(set(self.labels)), batch_size=self.clustering_batch_size
+            n_clusters=len(set(self.labels)), batch_size=self.clustering_batch_size, n_init="auto"
```

Comment:
Amazing work! To clarify didn't you want to set the value of `n_init` explicitly to 3 instead of `auto`? (https://github.com/embeddings-benchmark/mteb/issues/101#issue-1650808379)

Reply:
Good point! I based myself on [this snippet from the sklearn codebase](https://github.com/scikit-learn/scikit-learn/blob/9aaed4987/sklearn/cluster/_kmeans.py#L877).

The default initialization function for `MiniBatchKMeans` is `kmeans++`, which is the one you also select. If you select `auto`, then the number of inits will be based on the initialization function, and will be 1 for `kmeans++`, and 3 otherwise, suggesting that 3 is too high anyway. But I can change it to 3 for backwards compatibility of course.

Reply:
I see, so changing it to `auto` like in this PR would make new clustering scores different from previous ones, but is likely the better choice as it aligns with sklearn, correct?

Changing it to `3`, meanwhile, would ensure backwards compat, right?

Reply:
Yep, I think that's the choice we'll have to make.

Reply:
I see, I think breaking backwards compat is fine! I just ran it on Biorxiv w/ `average_word_embeddings_komninos` & it gets a tiny bit worse, but if it's what's recommended by sklearn let's go with this 👍

From:
```json
{
  "dataset_revision": "258694dd0231531bc1fd9de6ceb52a0853c6d908",
  "mteb_dataset_name": "BiorxivClusteringS2S",
  "mteb_version": "1.0.3.dev0",
  "test": {
    "evaluation_time": 10.24,
    "v_measure": 0.208173761965734,
    "v_measure_std": 0.006864121082528612
  }
}
```

to 

```json
{
  "dataset_revision": "258694dd0231531bc1fd9de6ceb52a0853c6d908",
  "mteb_dataset_name": "BiorxivClusteringS2S",
  "mteb_version": "1.0.3.dev0",
  "test": {
    "evaluation_time": 6.62,
    "v_measure": 0.20810107842666378,
    "v_measure_std": 0.008176457910048483
  }
}
```